src/pages/index.js

```diff
@@ -1,100 +1,259 @@
+  );
+  return card.getView(); // Return the card element to render
+};
+
+// Initialize the Section class to render and manage the card list in the UI
 const cardSection = new Section(
   {
-    items: initialCards, // Array of initial cards to render
     renderer: (cardData) => {
-      const cardElement = createCard(cardData); // Create the card and get the rendered card element
-      cardSection.addItem(cardElement);
+      const cardElement = createCard(cardData); // Create a card element for each card
+      cardSection.addItem(cardElement); // Add card to the card list (prepend it)
     },
   },
-  ".cards__list"
+  ".cards__list" // Container where cards will be appended
 );
 
-// Initialize the profile edit popup with form handling
+// Fetch and render user info and cards after page load
+let currentUserId; // Variable to store the current user's ID
+api
+  .getUserInfo()
+  .then((userInfo) => {
+    // Set user info on the page using the UserInfo instance
+    userInfoInstance.setUserInfo({
+      name: userInfo.name, // Set profile name
+      description: userInfo.about, // Set profile description
+      avatar: userInfo.avatar, // Set profile avatar
+    });
+    currentUserId = userInfo._id; // Store the current user ID
+
+    return api.getInitialCards(); // Fetch initial cards from the server
+  })
+  .then((cards) => {
+    // Render each card on the page
+    cards.forEach((cardData) => {
+      const cardElement = createCard({
+        name: cardData.name,
+        link: cardData.link,
+        _id: cardData._id,
+        owner: cardData.owner,
+        isLiked: cardData.isLiked,
+      });
+      cardSection.addItem(cardElement); // Add each card to the card section
+    });
+  })
+  .catch((err) => console.error("Error fetching user info or cards:", err)); // Handle errors during API requests
+
+// Opens profile edit popup and pre-fills the form with current user info
+const handleProfileEdit = () => {
+  const userData = userInfoInstance.getUserInfo(); // Get current profile data
+  profileTitleInput.value = userData.name; // Pre-fill profile name input
+  profileDescriptionInput.value = userData.description; // Pre-fill profile description
+  profileFormValidator.resetValidation(); // Reset form validation state
+  profileEditPopupInstance.open(); // Open the profile edit popup
+};
+
+// Opens the image popup when a card image is clicked
+const handleImagePopup = (link, name) => {
+  picturePopupInstance.open({ name, link }); // Open the popup with the image and its caption
+};
+
+// Initialize profile edit popup with form submission handling
 const profileEditPopupInstance = new PopupWithForm(
   "#profile-edit-popup",
   (formData) => {
-    userInfo.setUserInfo({
-      name: formData.name,
-      description: formData.description,
-    });
-    profileEditPopupInstance.close(); // Close the popup after submission
+    saveProfileButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```

Reply:
much better, thank you